src/main/java/org/truffleruby/core/regexp/InterpolatedRegexpNode.java

```diff
@@ -9,71 +9,115 @@
  */
 package org.truffleruby.core.regexp;
 
-import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
-import com.oracle.truffle.api.frame.VirtualFrame;
-import com.oracle.truffle.api.nodes.ExplodeLoop;
-import com.oracle.truffle.api.object.DynamicObject;
 import org.jcodings.specific.ASCIIEncoding;
 import org.jcodings.specific.USASCIIEncoding;
 import org.truffleruby.Layouts;
+import org.truffleruby.Log;
+import org.truffleruby.core.regexp.InterpolatedRegexpNodeFactory.RegexpBuilderNodeGen;
 import org.truffleruby.core.rope.Rope;
 import org.truffleruby.core.rope.RopeBuilder;
+import org.truffleruby.core.rope.RopeNodes;
 import org.truffleruby.core.rope.RopeOperations;
 import org.truffleruby.core.string.StringOperations;
 import org.truffleruby.language.RubyNode;
 import org.truffleruby.language.dispatch.CallDispatchHeadNode;
 import org.truffleruby.parser.BodyTranslator;
 
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.NodeChild;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.nodes.ExplodeLoop;
+import com.oracle.truffle.api.object.DynamicObject;
+
 public class InterpolatedRegexpNode extends RubyNode {
 
     @Children private final RubyNode[] children;
-    private final RegexpOptions options;
     @Child private CallDispatchHeadNode toS = CallDispatchHeadNode.create();
+    @Child private RegexpBuilderNode builderNode;
 
     public InterpolatedRegexpNode(RubyNode[] children, RegexpOptions options) {
         this.children = children;
-        this.options = options;
+        builderNode = RegexpBuilderNode.create(options);
     }
 
     @Override
     public Object execute(VirtualFrame frame) {
-        return createRegexp(executeChildren(frame));
+        return builderNode.execute(executeChildren(frame));
     }
 
-    @TruffleBoundary
-    private DynamicObject createRegexp(DynamicObject[] parts) {
-        final Rope[] strings = new Rope[children.length];
+    @ExplodeLoop
+    protected Rope[] executeChildren(VirtualFrame frame) {
+        Rope[] values = new Rope[children.length];
+        for (int i = 0; i < children.length; i++) {
+            final Object value = children[i].execute(frame);
+            values[i] = StringOperations.rope((DynamicObject) toS.call(frame, value, "to_s"));
+        }
+        return values;
+    }
+
+    @NodeChild(value = "ropes", type = RubyNode[].class)
```

Comment:
I think the type should still be just `RubyNode.class` as a single argument is passed, a `Rope[]`, not multiple arguments.